src/docbuilder/rustwide_builder.rs

```diff
@@ -648,6 +649,29 @@ impl RustwideBuilder {
         )
         .map(|t| t.1)
     }
+
+    fn should_build(
+        &self,
+        conn: &mut Client,
+        docbuilder: &DocBuilder,
+        name: &str,
+        version: &str,
+    ) -> Result<bool> {
+        if docbuilder.options().skip_if_exists {
+            // Check whether no successful builds are present in the database.
+            Ok(conn
+                .query(
+                    "SELECT 1 FROM crates, releases, builds
+                     WHERE crates.id = releases.crate_id AND releases.id = builds.rid
```

Comment:
This needs an `INNER JOIN` or you'll get duplicates.

Reply:
Hmm? This is working as expected when testing manually. Why would you expect it to fail?

Reply:
Huh, I expected it to fail when you have three different crates, each of which satisfies _one_ of these conditions.
```
CREATE TEMPORARY TABLE crates (
  id SERIAL PRIMARY KEY,
  name VARCHAR(255) UNIQUE NOT NULL
);

CREATE TEMPORARY TABLE releases (
   id SERIAL PRIMARY KEY,
   crate_id INT NOT NULL REFERENCES crates(id),
   version VARCHAR(100)
);

CREATE TEMPORARY TABLE builds (
   id SERIAL,
   rid INT NOT NULL REFERENCES releases(id),
   build_status BOOL NOT NULL
);

insert into crates (name) values ('a'), ('b');
insert into releases (crate_id, version) values (1, '0.1.0'), (2, '0.1.0');
insert into builds (rid, build_status) values (1, true), (2, true);

SELECT 1 FROM crates, releases, builds
WHERE crates.id = releases.crate_id AND releases.id = builds.rid
  AND crates.name = 'a' AND releases.version = '0.1.0'  AND builds.build_status = TRUE;
```
But that doesn't happen - the `WHERE id = crate_id AND id = rid` filters out the duplicates. It's more clear when you select everything instead of `1`:
```
select * from crates, releases, builds WHERE crates.id = releases.crate_id AND releases.id = builds.rid;
 id | name | id | crate_id | version | id | rid | build_status 
----+------+----+----------+---------+----+-----+--------------
  1 | a    |  1 |        1 | 0.1.0   |  1 |   1 | f
  2 | b    |  2 |        2 | 0.1.0   |  2 |   2 | t
joshua=> select * from crates, releases, builds WHERE crates.name = 'a' AND releases.version = '0.1.0'  AND builds.build_status = TRUE;
 id | name | id | crate_id | version | id | rid | build_status 
----+------+----+----------+---------+----+-----+--------------
  1 | a    |  1 |        1 | 0.1.0   |  2 |   2 | t
  1 | a    |  2 |        2 | 0.1.0   |  2 |   2 | t
```
So this _works_, it's just confusing.